Migrating a large portion of the V1 code base

DouglasPearless opened the pull request:
This brings across most of the V1 code base, except for:

- USB
- Ethernet
- SD Card

and the functions that require them.

DouglasPearless committed:
new file:   makefile
	new file:   src/Segger_SystemView/Global.h
	new file:   src/Segger_SystemView/SEGGER.h
	new file:   src/Segger_SystemView/SEGGER_RTT.c
	new file:   src/Segger_SystemView/SEGGER_RTT.h
	new file:   src/Segger_SystemView/SEGGER_RTT_Conf.h
	new file:   src/Segger_SystemView/SEGGER_SYSVIEW.c
	new file:   src/Segger_SystemView/SEGGER_SYSVIEW.h
	new file:   src/Segger_SystemView/SEGGER_SYSVIEW_Conf.h
	new file:   src/Segger_SystemView/SEGGER_SYSVIEW_ConfDefaults.h
	new file:   src/Segger_SystemView/SEGGER_SYSVIEW_Config_NoOS.c
	new file:   src/Segger_SystemView/SEGGER_SYSVIEW_Int.h
	new file:   src/config.default.cartesian
	new file:   src/generate-version.bat
	new file:   src/generate-version.sh
	new file:   src/libs/Debug/HardFault_HandlerC.c
	new file:   src/modules/tools/extruder/Extruder.cpp
	new file:   src/modules/tools/extruder/Extruder.h
	new file:   src/modules/tools/extruder/ExtruderMaker.cpp
	new file:   src/modules/tools/extruder/ExtruderMaker.h
	new file:   src/modules/tools/extruder/ExtruderPublicAccess.h
	new file:   src/modules/tools/filamentdetector/FilamentDetector.cpp
	new file:   src/modules/tools/filamentdetector/FilamentDetector.h
	new file:   src/modules/tools/rotarydeltacalibration/RotaryDeltaCalibration.cpp
	new file:   src/modules/tools/rotarydeltacalibration/RotaryDeltaCalibration.h
	new file:   src/modules/tools/scaracal/SCARAcal.cpp
	new file:   src/modules/tools/scaracal/SCARAcal.h
	new file:   src/modules/tools/spindle/Spindle.cpp
	new file:   src/modules/tools/spindle/Spindle.h
	new file:   src/modules/tools/temperatureswitch/TemperatureSwitch.cpp
	new file:   src/modules/tools/temperatureswitch/TemperatureSwitch.h
	new file:   src/modules/utils/PlayLed/PlayLed.cpp
	new file:   src/modules/utils/PlayLed/PlayLed.h
	new file:   src/modules/utils/configurator/Configurator.cpp
	new file:   src/modules/utils/configurator/Configurator.h
	new file:   src/modules/utils/currentcontrol/CurrentControl.cpp
	new file:   src/modules/utils/currentcontrol/CurrentControl.h
	new file:   src/modules/utils/currentcontrol/DigipotBase.h
	new file:   src/modules/utils/currentcontrol/ad5206.h
	new file:   src/modules/utils/currentcontrol/mcp4451.h
	new file:   src/modules/utils/killbutton/KillButton.cpp
	new file:   src/modules/utils/killbutton/KillButton.h
	new file:   src/modules/utils/motordrivercontrol/MotorDriverControl.cpp
	new file:   src/modules/utils/motordrivercontrol/MotorDriverControl.h
	new file:   src/modules/utils/motordrivercontrol/drivers/DRV8711/drv8711.cpp
	new file:   src/modules/utils/motordrivercontrol/drivers/DRV8711/drv8711.h

DouglasPearless committed:
modified:   src/config.default
	modified:   src/libs/Pin.cpp
	modified:   src/libs/Pin.h
	modified:   src/libs/StepTicker.cpp
	modified:   src/libs/StepperMotor.cpp
	modified:   src/libs/StepperMotor.h
	modified:   src/libs/checksumm.h
	modified:   src/libs/nuts_bolts.h
	modified:   src/libs/utils.h
	modified:   src/main.cpp
	modified:   src/makefile
	modified:   src/modules/robot/Planner.cpp
	modified:   src/modules/robot/Robot.cpp
	modified:   src/modules/robot/Robot.h
	modified:   src/modules/robot/Stepper.cpp
	modified:   src/modules/robot/arm_solutions/RotaryDeltaSolution.cpp
	modified:   src/modules/tools/endstops/Endstops.cpp
	modified:   src/modules/tools/temperaturecontrol/max31855.cpp

DouglasPearless committed:
modified:   src/config.default
	modified:   src/libs/Pin.cpp
	modified:   src/libs/Pin.h
	modified:   src/libs/StepTicker.cpp
	modified:   src/libs/StepperMotor.cpp
	modified:   src/libs/StepperMotor.h
	modified:   src/libs/checksumm.h
	modified:   src/libs/nuts_bolts.h
	modified:   src/libs/utils.h
	modified:   src/main.cpp
	modified:   src/makefile
	modified:   src/modules/robot/Planner.cpp
	modified:   src/modules/robot/Robot.cpp
	modified:   src/modules/robot/Robot.h
	modified:   src/modules/robot/Stepper.cpp
	modified:   src/modules/robot/arm_solutions/RotaryDeltaSolution.cpp
	modified:   src/modules/tools/endstops/Endstops.cpp
	modified:   src/modules/tools/temperaturecontrol/max31855.cpp

DouglasPearless committed:
modified:   README.md

DouglasPearless committed:
modified:   src/modules/robot/Stepper.cpp

DouglasPearless committed:
modified:   src/modules/tools/temperaturecontrol/max31855.cpp

DouglasPearless committed:
deleted:    src/Segger_SystemView/Global.h
	new file:   src/Segger_SystemView/README.md
	deleted:    src/Segger_SystemView/SEGGER.h
	deleted:    src/Segger_SystemView/SEGGER_RTT.c
	deleted:    src/Segger_SystemView/SEGGER_RTT.h
	deleted:    src/Segger_SystemView/SEGGER_RTT_Conf.h
	deleted:    src/Segger_SystemView/SEGGER_SYSVIEW.c
	deleted:    src/Segger_SystemView/SEGGER_SYSVIEW.h
	deleted:    src/Segger_SystemView/SEGGER_SYSVIEW_Conf.h
	deleted:    src/Segger_SystemView/SEGGER_SYSVIEW_ConfDefaults.h
	deleted:    src/Segger_SystemView/SEGGER_SYSVIEW_Config_NoOS.c
	deleted:    src/Segger_SystemView/SEGGER_SYSVIEW_Int.h

DouglasPearless committed:
modified:   src/Segger_SystemView/README.md

DouglasPearless committed:
modified:   src/Segger_SystemView/README.md

DouglasPearless committed:
modified:   src/Segger_SystemView/README.md

DouglasPearless committed:
modified:   src/Segger_SystemView/README.md

DouglasPearless committed:
modified:   src/Segger_SystemView/README.md

DouglasPearless committed:
modified:   libs/ADC/BurstADC.cpp
	modified:   libs/HeapRing.cpp
	modified:   libs/HeapRing.h
	modified:   libs/Kernel.cpp
	modified:   libs/Kernel.h
	modified:   libs/StepTicker.cpp
	modified:   libs/StepTicker.h
	modified:   libs/StepperMotor.cpp
	modified:   libs/StepperMotor.h
	modified:   modules/communication/GcodeDispatch.cpp
	modified:   modules/communication/utils/Gcode.cpp
	modified:   modules/communication/utils/Gcode.h
	modified:   modules/robot/Block.cpp
	modified:   modules/robot/Block.h
	modified:   modules/robot/Conveyor.cpp
	modified:   modules/robot/Conveyor.h
	modified:   modules/robot/Planner.cpp
	modified:   modules/robot/Planner.h
	modified:   modules/robot/Robot.cpp
	modified:   modules/robot/Robot.h
	deleted:    modules/robot/Stepper.cpp
	deleted:    modules/robot/Stepper.h
	modified:   modules/tools/drillingcycles/Drillingcycles.cpp
	modified:   modules/tools/endstops/Endstops.cpp
	modified:   modules/tools/endstops/Endstops.h
	modified:   modules/tools/filamentdetector/FilamentDetector.cpp
	modified:   modules/tools/laser/Laser.cpp
	modified:   modules/tools/laser/Laser.h
	modified:   modules/tools/spindle/Spindle.cpp
	modified:   modules/tools/switch/Switch.cpp
	modified:   modules/tools/temperaturecontrol/TemperatureControl.cpp
	modified:   modules/tools/temperaturecontrol/TemperatureControl.h
	modified:   modules/tools/temperaturecontrol/TemperatureControlPool.cpp
	modified:   modules/tools/temperaturecontrol/TemperatureControlPool.h
	modified:   modules/tools/temperaturecontrol/max31855.cpp
	modified:   modules/tools/temperaturecontrol/max31855.h
	modified:   modules/tools/temperaturecontrol/predefined_thermistors.h
	modified:   modules/tools/toolmanager/Tool.h
	modified:   modules/tools/toolmanager/ToolManager.cpp
	modified:   modules/utils/PlayLed/PlayLed.cpp
	modified:   modules/utils/currentcontrol/ad5206.h
	modified:   modules/utils/motordrivercontrol/drivers/TMC21X/TMC21X.cpp
	modified:   modules/utils/motordrivercontrol/drivers/TMC26X/TMC26X.cpp

DouglasPearless committed:
modified:   libs/ADC/BurstADC.cpp
	modified:   libs/HeapRing.cpp
	modified:   libs/HeapRing.h
	modified:   libs/Kernel.cpp
	modified:   libs/Kernel.h
	modified:   libs/StepTicker.cpp
	modified:   libs/StepTicker.h
	modified:   libs/StepperMotor.cpp
	modified:   libs/StepperMotor.h
	modified:   modules/communication/GcodeDispatch.cpp
	modified:   modules/communication/utils/Gcode.cpp
	modified:   modules/communication/utils/Gcode.h
	modified:   modules/robot/Block.cpp
	modified:   modules/robot/Block.h
	modified:   modules/robot/Conveyor.cpp
	modified:   modules/robot/Conveyor.h
	modified:   modules/robot/Planner.cpp
	modified:   modules/robot/Planner.h
	modified:   modules/robot/Robot.cpp
	modified:   modules/robot/Robot.h
	deleted:    modules/robot/Stepper.cpp
	deleted:    modules/robot/Stepper.h
	modified:   modules/tools/drillingcycles/Drillingcycles.cpp
	modified:   modules/tools/endstops/Endstops.cpp
	modified:   modules/tools/endstops/Endstops.h
	modified:   modules/tools/filamentdetector/FilamentDetector.cpp
	modified:   modules/tools/laser/Laser.cpp
	modified:   modules/tools/laser/Laser.h
	modified:   modules/tools/spindle/Spindle.cpp
	modified:   modules/tools/switch/Switch.cpp
	modified:   modules/tools/temperaturecontrol/TemperatureControl.cpp
	modified:   modules/tools/temperaturecontrol/TemperatureControl.h
	modified:   modules/tools/temperaturecontrol/TemperatureControlPool.h
	modified:   modules/tools/temperaturecontrol/max31855.cpp
	modified:   modules/tools/temperaturecontrol/max31855.h
	modified:   modules/tools/temperaturecontrol/predefined_thermistors.h
	modified:   modules/tools/toolmanager/Tool.h
	modified:   modules/tools/toolmanager/ToolManager.cpp
	modified:   modules/utils/PlayLed/PlayLed.cpp
	modified:   modules/utils/currentcontrol/ad5206.h
	modified:   modules/utils/motordrivercontrol/drivers/TMC21X/TMC21X.cpp
	modified:   modules/utils/motordrivercontrol/drivers/TMC26X/TMC26X.cpp

DouglasPearless committed:
modified:   modules/tools/temperaturecontrol/TemperatureControlPool.cpp

DouglasPearless committed:
modified:   ../README.md

DouglasPearless committed:
modified:   ../README.md

DouglasPearless committed:
modified:   libs/Kernel.cpp

DouglasPearless committed:
modified:   libs/Adc.cpp
	modified:   libs/RingBuffer.h
	modified:   libs/utils.cpp
	modified:   libs/utils.h
	modified:   libs/nuts_bolts.h

DouglasPearless committed:
new file:   libs/md5.h

DouglasPearless committed:
modified:   libs/nuts_bolts.h

DouglasPearless committed:
modified:   libs/Kernel.cpp
	modified:   libs/Kernel.h
	modified:   modules/utils/motordrivercontrol/MotorDriverControl.cpp

DouglasPearless committed:
modified:   libs/Kernel.cpp

DouglasPearless committed:
modified:   gcc4mbed/external/mbed/libraries/mbed/hal/sleep_api.h
	modified:   makefile
	modified:   src/libs/Hook.h
	modified:   src/modules/communication/GcodeDispatch.cpp
	modified:   src/modules/communication/SerialConsole.cpp
	modified:   src/modules/communication/SerialConsole.h
	modified:   src/modules/robot/ActuatorCoordinates.h
	modified:   src/modules/robot/Robot.h
	modified:   src/modules/robot/arm_solutions/BaseSolution.h
	modified:   src/modules/robot/arm_solutions/CartesianSolution.cpp
	modified:   src/modules/robot/arm_solutions/CartesianSolution.h
	modified:   src/modules/robot/arm_solutions/CoreXZSolution.cpp
	modified:   src/modules/robot/arm_solutions/CoreXZSolution.h
	modified:   src/modules/robot/arm_solutions/ExperimentalDeltaSolution.cpp
	modified:   src/modules/robot/arm_solutions/ExperimentalDeltaSolution.h
	modified:   src/modules/robot/arm_solutions/HBotSolution.cpp
	modified:   src/modules/robot/arm_solutions/HBotSolution.h
	modified:   src/modules/robot/arm_solutions/LinearDeltaSolution.cpp
	modified:   src/modules/robot/arm_solutions/LinearDeltaSolution.h
	modified:   src/modules/robot/arm_solutions/MorganSCARASolution.cpp
	modified:   src/modules/robot/arm_solutions/MorganSCARASolution.h
	modified:   src/modules/robot/arm_solutions/RotaryDeltaSolution.cpp
	modified:   src/modules/robot/arm_solutions/RotaryDeltaSolution.h
	modified:   src/modules/robot/arm_solutions/RotatableCartesianSolution.cpp
	modified:   src/modules/robot/arm_solutions/RotatableCartesianSolution.h
	modified:   src/modules/tools/endstops/Endstops.cpp
	modified:   src/modules/tools/extruder/Extruder.cpp
	modified:   src/modules/tools/rotarydeltacalibration/RotaryDeltaCalibration.cpp
	modified:   src/modules/tools/scaracal/SCARAcal.cpp
	modified:   src/modules/tools/spindle/Spindle.cpp
	modified:   src/modules/tools/switch/Switch.cpp
	modified:   src/modules/tools/temperaturecontrol/max31855.cpp
	modified:   src/modules/tools/temperaturecontrol/predefined_thermistors.h
	modified:   src/modules/tools/toolmanager/Tool.h
	new file:   src/modules/tools/zprobe/DeltaCalibrationStrategy.cpp
	new file:   src/modules/tools/zprobe/DeltaCalibrationStrategy.h
	new file:   src/modules/tools/zprobe/DeltaGridStrategy.cpp
	new file:   src/modules/tools/zprobe/DeltaGridStrategy.h
	new file:   src/modules/tools/zprobe/LevelingStrategy.h
	new file:   src/modules/tools/zprobe/Plane3D.cpp
	new file:   src/modules/tools/zprobe/Plane3D.h
	new file:   src/modules/tools/zprobe/ThreePointStrategy.cpp
	new file:   src/modules/tools/zprobe/ThreePointStrategy.h
	new file:   src/modules/tools/zprobe/ZGridStrategy.cpp
	new file:   src/modules/tools/zprobe/ZGridStrategy.h
	new file:   src/modules/tools/zprobe/ZProbe.cpp
	new file:   src/modules/tools/zprobe/ZProbe.h
	modified:   src/modules/utils/motordrivercontrol/drivers/TMC26X/TMC26X.cpp

DouglasPearless committed:
modified:   src/main.cpp
	modified:   src/makefile

DouglasPearless committed:
modified:   libs/Kernel.cpp
	modified:   libs/Kernel.h
	modified:   libs/StepTicker.cpp
	modified:   libs/StepTicker.h
	modified:   libs/StepperMotor.cpp
	modified:   libs/StepperMotor.h
	modified:   makefile
	modified:   modules/communication/GcodeDispatch.cpp
	modified:   modules/communication/utils/Gcode.cpp
	modified:   modules/communication/utils/Gcode.h
	modified:   modules/robot/ActuatorCoordinates.h
	modified:   modules/robot/Block.cpp
	modified:   modules/robot/Block.h
	modified:   modules/robot/Robot.cpp
	modified:   modules/robot/Robot.h
	modified:   modules/robot/arm_solutions/BaseSolution.h
	modified:   modules/robot/arm_solutions/CartesianSolution.cpp
	modified:   modules/robot/arm_solutions/CartesianSolution.h
	modified:   modules/robot/arm_solutions/CoreXZSolution.cpp
	modified:   modules/robot/arm_solutions/CoreXZSolution.h
	modified:   modules/robot/arm_solutions/ExperimentalDeltaSolution.cpp
	modified:   modules/robot/arm_solutions/ExperimentalDeltaSolution.h
	modified:   modules/robot/arm_solutions/HBotSolution.cpp
	modified:   modules/robot/arm_solutions/HBotSolution.h
	modified:   modules/robot/arm_solutions/LinearDeltaSolution.cpp
	modified:   modules/robot/arm_solutions/LinearDeltaSolution.h
	modified:   modules/robot/arm_solutions/MorganSCARASolution.cpp
	modified:   modules/robot/arm_solutions/MorganSCARASolution.h
	modified:   modules/robot/arm_solutions/RotaryDeltaSolution.cpp
	modified:   modules/robot/arm_solutions/RotaryDeltaSolution.h
	modified:   modules/robot/arm_solutions/RotatableCartesianSolution.cpp
	modified:   modules/robot/arm_solutions/RotatableCartesianSolution.h
	modified:   modules/tools/drillingcycles/Drillingcycles.cpp
	modified:   modules/tools/drillingcycles/Drillingcycles.h

DouglasPearless committed:
modified:   ../README.md
	modified:   libs/Kernel.cpp
	modified:   libs/StepTicker.cpp
	modified:   makefile
	modified:   modules/robot/Planner.cpp

DouglasPearless committed:
modified:   ../README.md

DouglasPearless committed:
modified:   main.cpp

DouglasPearless committed:
modified:   ../README.md

DouglasPearless committed:
modified:   ../README.md
	modified:   libs/Debug/HardFault_HandlerC.c
	modified:   main.cpp

DouglasPearless committed:
modified:   libs/Debug/HardFault_HandlerC.c

DouglasPearless committed:
modified:   README.md
	modified:   src/config.default
	modified:   src/modules/robot/Robot.cpp

DouglasPearless committed:
modified:   src/modules/robot/Robot.cpp

DouglasPearless committed:
modified:   config.default
	modified:   libs/StepTicker.cpp
	modified:   modules/robot/Planner.cpp

DouglasPearless committed:
modified:   ../README.md

DouglasPearless committed:
modified:   libs/SlowTicker.cpp
	modified:   libs/SlowTicker.h

DouglasPearless committed:
modified:   ../README.md

DouglasPearless committed:
modified:   ../README.md

DouglasPearless committed:
modified:   libs/Kernel.cpp
	modified:   makefile

DouglasPearless committed:
modified:   libs/Kernel.cpp
	modified:   libs/SlowTicker.cpp
	modified:   libs/StepTicker.cpp
	modified:   main.cpp
	modified:   makefile

DouglasPearless committed:
Merge branch 'master' of https://github.com/Smoothieware/Smoothie2

	modified:   ../README.md
	modified:   main.cpp

DouglasPearless committed:
modified:   ../README.md
	modified:   generate-version.sh
	modified:   libs/Kernel.cpp
	modified:   libs/SlowTicker.cpp
	modified:   libs/StepTicker.cpp
	modified:   main.cpp
	modified:   makefile

DouglasPearless committed:
modified:   config.default

DouglasPearless committed:
modified:   main.cpp

DouglasPearless committed:
modified:   README.md

DouglasPearless committed:
Merge branch 'master' of https://github.com/DouglasPearless/Smoothie2

DouglasPearless committed:
modified:   README.md

DouglasPearless committed:
modified:   README.md

DouglasPearless committed:
modified:   config.default
	modified:   libs/Kernel.cpp
	new file:   libs/MemoryPool.cpp
	new file:   libs/MemoryPool.h
	modified:   libs/SlowTicker.cpp
	modified:   libs/StepTicker.cpp
	modified:   libs/StepTicker.h
	new file:   libs/platform_memory.cpp
	new file:   libs/platform_memory.h
	modified:   modules/tools/endstops/Endstops.cpp
	modified:   modules/tools/laser/Laser.cpp
	modified:   modules/tools/laser/Laser.h

DouglasPearless committed:
modified:   src/libs/Kernel.cpp
	modified:   src/libs/Kernel.h
	new file:   src/libs/MemoryPool.cpp
	new file:   src/libs/MemoryPool.h
	modified:   src/libs/Pwm.cpp
	modified:   src/libs/RingBuffer.h
	modified:   src/libs/SlowTicker.cpp
	modified:   src/libs/SlowTicker.h
	modified:   src/libs/StepTicker.cpp
	modified:   src/libs/StepTicker.h
	modified:   src/libs/StepperMotor.cpp
	modified:   src/libs/StepperMotor.h
	modified:   src/libs/nuts_bolts.h
	new file:   src/libs/platform_memory.cpp
	new file:   src/libs/platform_memory.h
	modified:   src/libs/utils.cpp
	modified:   src/libs/utils.h
	modified:   src/main.cpp
	modified:   src/makefile
	modified:   src/modules/robot/ActuatorCoordinates.h
	modified:   src/modules/robot/Block.h
	modified:   src/modules/robot/Conveyor.h
	modified:   src/modules/robot/Planner.cpp
	modified:   src/modules/tools/endstops/Endstops.cpp
	modified:   src/modules/tools/endstops/Endstops.h
	modified:   src/modules/tools/laser/Laser.cpp
	modified:   src/modules/tools/laser/Laser.h
	modified:   src/modules/tools/switch/Switch.cpp

DouglasPearless committed:
modified:   src/config.default
	modified:   src/modules/tools/spindle/Spindle.cpp

DouglasPearless committed:
Merge branch 'master' into HEAD

* master:
  modified:   config.default 	modified:   libs/Kernel.cpp 	new file:   libs/MemoryPool.cpp 	new file:   libs/MemoryPool.h 	modified:   libs/SlowTicker.cpp 	modified:   libs/StepTicker.cpp 	modified:   libs/StepTicker.h 	new file:   libs/platform_memory.cpp 	new file:   libs/platform_memory.h 	modified:   modules/tools/endstops/Endstops.cpp 	modified:   modules/tools/laser/Laser.cpp 	modified:   modules/tools/laser/Laser.h

Conflicts:
	src/config.default
	src/libs/Kernel.cpp
	src/libs/SlowTicker.cpp
	src/libs/StepTicker.cpp

	modified:   src/config.default
	modified:   src/libs/Kernel.cpp
	modified:   src/libs/SlowTicker.cpp
	modified:   src/libs/StepTicker.cpp
	modified:   src/modules/tools/endstops/Endstops.cpp

DouglasPearless committed:
modified:   src/modules/tools/endstops/Endstops.cpp

DouglasPearless committed:
modified:   src/modules/tools/endstops/Endstops.cpp

DouglasPearless committed:
modified:   src/Segger_SystemView/README.md

DouglasPearless committed:
modified:   src/Segger_SystemView/README.md

DouglasPearless committed:
modified:   .gitignore

DouglasPearless committed:
modified:   libs/ADC/BurstADC.cpp
	modified:   libs/Adc.cpp
	modified:   libs/Kernel.cpp
	modified:   libs/Kernel.h
	new file:   libs/MemoryPool.cpp
	new file:   libs/MemoryPool.h
	modified:   libs/Pin.cpp
	modified:   libs/Pin.h
	modified:   libs/RingBuffer.h
	modified:   libs/SlowTicker.cpp
	modified:   libs/SlowTicker.h
	modified:   libs/StepTicker.cpp
	modified:   libs/StepTicker.h
	modified:   libs/StepperMotor.cpp
	modified:   libs/StepperMotor.h
	modified:   libs/nuts_bolts.h
	new file:   libs/platform_memory.cpp
	new file:   libs/platform_memory.h
	modified:   libs/utils.cpp
	modified:   libs/utils.h
	modified:   modules/communication/GcodeDispatch.cpp
	modified:   modules/robot/Block.h
	modified:   modules/robot/Conveyor.h
	modified:   modules/robot/Planner.cpp
	modified:   modules/robot/Robot.cpp
	modified:   modules/tools/endstops/Endstops.cpp
	modified:   modules/tools/endstops/Endstops.h
	modified:   modules/tools/laser/Laser.cpp
	modified:   modules/tools/laser/Laser.h
	modified:   modules/tools/spindle/Spindle.cpp
	modified:   modules/tools/switch/Switch.cpp

DouglasPearless committed:
Merge branch 'master' of https://github.com/DouglasPearless/Smoothie2

	src/libs/Kernel.cpp
	src/libs/SlowTicker.cpp
	src/libs/StepTicker.cpp
	src/modules/tools/spindle/Spindle.cpp

	modified:   README.md

DouglasPearless committed:
modified:   src/main.cpp

DouglasPearless committed:
modified:   src/main.cpp

DouglasPearless committed:
modified:   src/libs/SlowTicker.cpp

DouglasPearless committed:
Merge branch 'master' of https://github.com/Smoothieware/Smoothie2

* 'master' of https://github.com/Smoothieware/Smoothie2:
  Updated welcome message with version generation
  Endstops working
  Changed period time calculations
  Changed base freq
  Motors Mooving... slowly
  Continuing adding changes from v1 step generation
  Conmtinuing adding changes from v1 step generation
  v1 files port fixes
  Initial port from v1 firmware
  Inserted Smoothie1 files

Conflicts:
	src/config.default
	src/generate-version.sh
	src/libs/Kernel.cpp
	src/libs/SlowTicker.cpp
	src/libs/StepTicker.cpp
	src/main.cpp
	src/makefile
	src/modules/robot/ActuatorCoordinates.h
	src/modules/tools/endstops/Endstops.cpp
	src/modules/tools/scaracal/SCARAcal.h
	src/modules/tools/toolmanager/ToolManager.cpp

DouglasPearless committed:
Update .gitignore

DouglasPearless committed:
modified:   makefile
	modified:   src/libs/Kernel.cpp
	modified:   src/main.cpp
	modified:   src/makefile
	modified:   src/modules/tools/endstops/Endstops.cpp

DouglasPearless committed:
modified:   modules/tools/endstops/Endstops.cpp

DouglasPearless committed:
Merge branch 'master' of https://github.com/DouglasPearless/Smoothie2

wolfmanjm commented:
it seems there is a bunch of stuff in here specific to your local directory/workspace. The Segger stuff needs to not be in the main repo IMO, and the .gitignore should not have your personal preferences for what to ignore.

DouglasPearless commented:
Hi Jim,

I agree; the .gitignore needs to be itself ignore and not have my stuff in it; will fix.

I will update the makefile to allow a flag for debug and segger so it is only include if the user wants it or there; what do you think?

Sent from my iPhone

> On 12/11/2016, at 7:30 PM, Jim Morris <notifications@github.com> wrote:
> 
> it seems there is a bunch of stuff in here specific to your local directory/workspace. The Segger stuff needs to not be in the main repo IMO, and the .gitignore should not have your personal preferences for what to ignore.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

wolfmanjm commented:
IMO there should be NO Segger stuff at all in a public open source project. that is private to you and should not be in the makefile or anywhere else publicly. This is true of any personal preferences.

DouglasPearless committed:
modified:   src/config.default

DouglasPearless commented:
Hi Jim,

Good point, instead it might be better I write up a "how to" like the Eclipse for the Smoothie website.

I have noticed strange behaviour for my Rotary Delta, specifically the mechanism always goes in the opposite direction for G0 moves; I have read the changes you made for the mirror_xy feature in the code, but that doesn't seem to be it; I have been looking for the cause for the last day; do you have any  thoughts of what it might be with all the code changes recently?

( just had a large earthquake here in New Zealand, certainly nice to focus on something else for a while as the city where I live - Wellington, has been shutdown )

Sent from my iPhone

> On 13/11/2016, at 8:19 AM, Jim Morris <notifications@github.com> wrote:
> 
> IMO there should be NO Segger stuff at all in a public open source project. that is private to you and should not be in the makefile or anywhere else publicly. This is true of any personal preferences.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

DouglasPearless commented:
Yes,

Like http://smoothieware.org/eclipse <http://smoothieware.org/eclipse> but for the Segger SystemView and how to get into tracing ISRs, Timers, interrupts etc to see what Smoothie is doing internally.

Cheers
Douglas
> On 14/11/2016, at 9:34 AM, Arthur Wolf <notifications@github.com> wrote:
> 
> On Sun, Nov 13, 2016 at 9:30 PM, Douglas Pearless <notifications@github.com>
> wrote:
> 
> > Hi Jim,
> >
> > Good point, instead it might be better I write up a "how to" like the
> > Eclipse for the Smoothie website.
> >
> 
> http://smoothieware.org/eclipse ?
> 
> 
> >
> > I have noticed strange behaviour for my Rotary Delta, specifically the
> > mechanism always goes in the opposite direction for G0 moves; I have read
> > the changes you made for the mirror_xy feature in the code, but that
> > doesn't seem to be it; I have been looking for the cause for the last day;
> > do you have any thoughts of what it might be with all the code changes
> > recently?
> >
> > ( just had a large earthquake here in New Zealand, certainly nice to focus
> > on something else for a while as the city where I live - Wellington, has
> > been shutdown )
> >
> > Sent from my iPhone
> >
> > > On 13/11/2016, at 8:19 AM, Jim Morris <notifications@github.com> wrote:
> > >
> > > IMO there should be NO Segger stuff at all in a public open source
> > project. that is private to you and should not be in the makefile or
> > anywhere else publicly. This is true of any personal preferences.
> > >
> > > —
> > > You are receiving this because you authored the thread.
> >
> > > Reply to this email directly, view it on GitHub, or mute the thread.
> > >
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/Smoothieware/Smoothie2/pull/36#issuecomment-260210372>,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/AAGpFeIKrv7hkj9uWJch-Pa5vO57d5EKks5q93N3gaJpZM4KvnT2>
> > .
> >
> 
> 
> 
> -- 
> Courage et bonne humeur.
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/Smoothieware/Smoothie2/pull/36#issuecomment-260210627>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ADkh-5phMdyOk58BwijcRIZvFw5ATQHQks5q93Q8gaJpZM4KvnT2>.
>

DouglasPearless committed:
modified:   src/config.default
	new file:   src/copy.default.rotarydelta.default
	modified:   src/libs/Kernel.cpp
	modified:   src/modules/robot/arm_solutions/RotaryDeltaSolution.cpp
	modified:   src/modules/tools/endstops/Endstops.cpp

DouglasPearless committed:
modified:   src/modules/robot/Planner.cpp
	modified:   src/modules/robot/arm_solutions/RotaryDeltaSolution.cpp
	modified:   src/modules/robot/arm_solutions/RotaryDeltaSolution.h
	modified:   src/modules/tools/endstops/Endstops.cpp

DouglasPearless committed:
modified:   src/modules/robot/Block.h
	modified:   src/modules/robot/Conveyor.h
	modified:   src/modules/tools/endstops/Endstops.cpp
	modified:   src/modules/tools/endstops/Endstops.h
	modified:   src/modules/tools/zprobe/ZProbe.cpp
